src/core/link.go

```diff
@@ -257,7 +257,7 @@ func (l *links) add(u *url.URL, sintf string, linkType linkType) error {
 				}
 
 				conn, err := l.connect(state.ctx, u, info, options)
-				if err != nil {
+				if err != nil || conn == nil {
```

Comment:
Since a nil error and a nil conn is weird, we should probably check if conn == nil and add some additional logging, then ask someone who sees the error to run it. E.g. at least see which protocol the peers are trying to connect over (maybe it's some edge case in the QUIC code?).

Reply:
Added this.